apps/etterlatte-beregning-kafka/src/main/kotlin/no/nav/etterlatte/beregningkafka/SjekkOmOverstyrtBeregningRiver.kt

```diff
@@ -0,0 +1,53 @@
+package no.nav.etterlatte.beregningkafka
+
+import io.ktor.http.HttpStatusCode
+import no.nav.etterlatte.libs.common.rapidsandrivers.setEventNameForHendelseType
+import no.nav.etterlatte.rapidsandrivers.BEHANDLING_ID_KEY
+import no.nav.etterlatte.rapidsandrivers.HENDELSE_DATA_KEY
+import no.nav.etterlatte.rapidsandrivers.Kontekst
+import no.nav.etterlatte.rapidsandrivers.ListenerMedLoggingOgFeilhaandtering
+import no.nav.etterlatte.rapidsandrivers.ReguleringHendelseType
+import no.nav.etterlatte.rapidsandrivers.TILBAKESTILTE_BEHANDLINGER_KEY
+import no.nav.etterlatte.rapidsandrivers.tilbakestilteBehandlinger
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import org.slf4j.LoggerFactory
+
+class SjekkOmOverstyrtBeregningRiver(
+    rapidsConnection: RapidsConnection,
+    private val beregningService: BeregningService,
+) : ListenerMedLoggingOgFeilhaandtering() {
+    private val logger = LoggerFactory.getLogger(this::class.java)
+
+    init {
+        initialiserRiver(rapidsConnection, ReguleringHendelseType.LOEPENDE_YTELSE_FUNNET) {
+            validate { it.rejectKey(BEHANDLING_ID_KEY) }
+            validate { it.requireKey(HENDELSE_DATA_KEY) }
+            validate { it.interestedIn(TILBAKESTILTE_BEHANDLINGER_KEY) }
+        }
+    }
+
+    override fun kontekst() = Kontekst.REGULERING
+
+    override fun haandterPakke(
+        packet: JsonMessage,
+        context: MessageContext,
+    ) {
+        logger.info("Mottatt sjekk om finnes åpen behandling med overstyrt beregning hendelse")
+
+        packet.tilbakestilteBehandlinger.forEach {
+            val overstyrt = beregningService.hentOverstyrt(it)
+            if (overstyrt.status != HttpStatusCode.NoContent) {
```

Comment:
Denne kjens litt risikabel for om vi får feilande statusar, men dei kastar kanskje exception uansett sånn at vi ikkje kjem fram hit, så da bør det jo funke.

Reply:
Ja jeg jeg tenkte slik
Hvis no NoContent: Vi vet at det ikke er overstyrt. Fortsetter.
Hvis OK: Vi vet at det er overstyrt kast feil.
Hvis noen annen status: Vi har ikke fått bekreftet at det ikke er overstyrt. Kast feil.

Men kan være greit med en egen exception for siste fordi nå kan vi gi feil indikasjon på at det er overstyrt.

Reply:
Gjorde dette i stedet
`

       packet.tilbakestilteBehandlinger.forEach {
            val overstyrt = beregningService.hentOverstyrt(it)
            when (overstyrt.status) {
                HttpStatusCode.NoContent -> return@forEach
                HttpStatusCode.OK -> throw KanIkkeRegulereSakMedAapenBehandlingOverstyrtBeregning()
                else -> throw KanIkkeBekrefteAtSakIkkeHarOverstyrtBeregning()
            }
        }
`

Reply:
Eksplisitt og fin!